server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -91,13 +91,19 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
         params.getDruidCluster().getCluster().entrySet()) {
       String tier = entry.getKey();
 
-      if (currentlyMovingSegments.get(tier) == null) {
-        currentlyMovingSegments.put(tier, new ConcurrentHashMap<String, BalancerSegmentHolder>());
-      }
+      currentlyMovingSegments.putIfAbsent(tier, new ConcurrentHashMap<>());
```

Comment:
map.computeIfAbsent() FTW. Plus the result of this call could be used in the next line, instead of calling `currentlyMovingSegments.get(tier)` again.